ledger-service/cli-opts/src/main/scala/cliopts/Logging.scala

```diff
@@ -7,6 +7,29 @@ import ch.qos.logback.classic.{Level => LogLevel}
 
 object Logging {
 
+  def reconfigure(clazz: Class[_], pathToLogbackFileInJarFile: String): Unit = {
+    // Try reconfiguring the library
+    import ch.qos.logback.core.joran.spi.JoranException
+    import ch.qos.logback.classic.LoggerContext
+    import ch.qos.logback.classic.joran.JoranConfigurator
+    import org.slf4j.LoggerFactory
+    val stream = clazz.getClassLoader.getResource(pathToLogbackFileInJarFile).openStream()
```

Comment:
You should be able to throw `Using`/try with resources at this.

Reply:
can you elaborate?

Reply:
```
diff --git a/ledger-service/cli-opts/src/main/scala/cliopts/Logging.scala b/ledger-service/cli-opts/src/main/scala/cliopts/Logging.scala
index 844fe711dc..d42b85a070 100644
--- a/ledger-service/cli-opts/src/main/scala/cliopts/Logging.scala
+++ b/ledger-service/cli-opts/src/main/scala/cliopts/Logging.scala
@@ -4,6 +4,7 @@
 package com.daml.cliopts

 import ch.qos.logback.classic.{Level => LogLevel}
+import scala.util.Using

 object Logging {

@@ -13,20 +14,20 @@ object Logging {
     import ch.qos.logback.classic.LoggerContext
     import ch.qos.logback.classic.joran.JoranConfigurator
     import org.slf4j.LoggerFactory
-    val stream = clazz.getClassLoader.getResource(pathToLogbackFileInJarFile).openStream()
-    try {
-      val context = LoggerFactory.getILoggerFactory.asInstanceOf[LoggerContext]
-      val configurator = new JoranConfigurator
-      configurator.setContext(context)
-      context.reset()
-      configurator.doConfigure(stream)
-    } catch {
-      case je: JoranException =>
-        // Fallback to System.err.println because the logger won't work in any way anymore.
-        System.err.println(s"reconfigured failed using url $pathToLogbackFileInJarFile: $je")
-        je.printStackTrace(System.err)
-    } finally {
-      stream.close()
+    Using.resource(clazz.getClassLoader.getResource(pathToLogbackFileInJarFile).openStream()) {
+      stream =>
+        try {
+          val context = LoggerFactory.getILoggerFactory.asInstanceOf[LoggerContext]
+          val configurator = new JoranConfigurator
+          configurator.setContext(context)
+          context.reset()
+          configurator.doConfigure(stream)
+        } catch {
+          case je: JoranException =>
+            // Fallback to System.err.println because the logger won't work in any way anymore.
+            System.err.println(s"reconfigured failed using url $pathToLogbackFileInJarFile: $je")
+            je.printStackTrace(System.err)
+        }
     }
   }
```
Definitely subjective but I like it because it makes it very explicit that you’re not leaking resources.

Reply:
I was looking for something which ensures that I'm not leaking resources, thanks a lot!